src/sentry/static/sentry/app/components/gridEditable/grid.tsx

```diff
@@ -0,0 +1,283 @@
+
+    return (
+      <GridHead>
+        <GridRow>
+          {/* GridHeadEdit must come first.
+
+              It is a <th> that uses `position: absolute` to set its placement.
+              The CSS selectors captures the last GridHeadCell and put a
+              padding-right to provide space for GridHeadEdit to be displayed.
+
+              FAQ:
+              Instead of using `position: absolute`, why can't we just put
+              GridHeadEdit at the end so it appears on the right?
+              Because CSS Grids need to have the same number of Head/Body cells
+              for everything to align properly. Sub-grids are new and may not be
+              well supported in older browsers/
+
+              Why can't we just put GridHeadEdit somewhere else?
+              Because HTML rules mandate that <div> cannot be a nested child of
+              a <table>. This seems the best way to make it correct to satisfy
+              HTML semantics. */
+          isEditable && this.renderGridHeadEditButtons()}
+
+          {columnOrder.map((column, i) => (
+            <GridHeadCell
+              key={`${column.key}${i}`}
+              isPrimary={column.isPrimary}
+              isEditing={isEditing}
+              actions={{
+                deleteColumn: () => actions.deleteColumn(i),
+                toggleModalEditColumn: this.toggleModalEditColumn,
+              }}
+            >
+              {grid.renderHeaderCell ? grid.renderHeaderCell(column) : column.name}
+            </GridHeadCell>
+          ))}
+        </GridRow>
+      </GridHead>
+    );
+  };
+
+  renderGridHeadEditButtons = () => {
+    if (!this.props.isEditable) {
+      return null;
+    }
+
+    if (!this.state.isEditing) {
+      return (
+        <GridEditGroup>
+          <GridEditGroupButton onClick={this.toggleEdit}>
+            <ToolTip title={t('Edit Columns')}>
+              <InlineSvg src="icon-edit-pencil" />
+            </ToolTip>
+          </GridEditGroupButton>
+        </GridEditGroup>
+      );
+    }
+
+    return (
+      <GridEditGroup>
+        <GridEditGroupButton onClick={this.toggleModalEditColumn}>
+          <ToolTip title={t('Add Columns')}>
+            <InlineSvg src="icon-circle-add" />
+          </ToolTip>
+        </GridEditGroupButton>
+        <GridEditGroupButton onClick={this.toggleEdit}>
+          <ToolTip title={t('Cancel Edit')}>
+            <InlineSvg src="icon-close" />
+          </ToolTip>
+        </GridEditGroupButton>
+      </GridEditGroup>
+    );
+  };
+
+  renderGridBody = () => {
+    const {data} = this.props;
+
+    if (!data || data.length === 0) {
+      return this.renderEmptyData();
+    }
+
+    return <GridBody>{data.map(this.renderGridBodyRow)}</GridBody>;
+  };
+
+  renderGridBodyRow = (dataRow: DataRow, row: number) => {
+    const {grid} = this.props;
+
+    return (
+      <GridRow key={row}>
+        {this.props.columnOrder.map((col, i) => (
+          <GridBodyCell key={`${col.name}${i}`}>
```

Comment:
Do we need the index? are column names unique?

Reply:
To keep things simple, there is no logic to stop a user from adding 2 of the same column into the grid, so having the index prevents wonky things from happening.

Reply:
I switched it to `${col.key}${i}`.